geopandas/io/parquet.py

```diff
@@ -0,0 +1,228 @@
+import json
+
+from pandas.io.common import get_filepath_or_buffer
+from pandas.io.parquet import PyArrowImpl as PandasPyArrowImpl
+from pandas import DataFrame
+
+from geopandas.array import from_wkb, to_wkb
+from geopandas import GeoDataFrame
+import geopandas
+
+
+PARQUET_METADATA_VERSION = "1.0.0"
+
+
+def _encode_crs(crs):
+    if isinstance(crs, str):
+        return {"proj4": crs}
+
+    return crs
+
+
+def _decode_crs(crs):
+    if crs and "proj4" in crs:
+        return crs["proj4"]
+
+    return crs
+
+
+class PyArrowImpl(PandasPyArrowImpl):
```

Comment:
One thing I am wondering: is it worth to subclass this? (it's also kind of a private pandas thing) 
What we get from it: the `validate_dataframe`, and the import checking, I think? (it does make it more convenient ..)

Reply:
We do the import checking ourselves (copied from `pandas`).  Originally, I was modeling the implementation on that of Pandas, so that it could be subclassed for `fastparquet`.

Since `fastparquet` doesn't support what we need, we could simplify our implementation here down to just the read / write functions and not use a class-based approach at all.  Would you prefer that?

Reply:
> Since fastparquet doesn't support what we need, we could simplify our implementation here down to just the read / write functions and not use a class-based approach at all. Would you prefer that?

Yes, for now that seems better then I think